types/metadata.go

```diff
@@ -61,6 +63,10 @@ func NewMetadataV10() *Metadata {
 	return &Metadata{Version: 10, IsMetadataV10: true, AsMetadataV10: MetadataV10{make([]ModuleMetadataV10, 0)}}
 }
 
+func NewMetadataV11() *Metadata {
+	return &Metadata{Version: 11, IsMetadataV11: true, AsMetadataV11: MetadataV11{}}
```

Comment:
Please create an empty slice above instead of keeping at as a null pointer, see MetadataV10

Reply:
That's not entirely true, they behave the same, but the are not the same: the nil slice means there is no underlying array for the slice. 

This has bitten me in the past since the scale codec always creates empty slices (and not nil slices) – so comparing two metadata instances will fail when encoding and decoding.

For consistency, please change this. Thanks!